src/installer/managed/Microsoft.NET.HostModel/AppHost/ElfUtils.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+
+namespace Microsoft.NET.HostModel.AppHost
+{
+    internal static class ElfUtils
+    {
+        // The Linux Headers are copied from elf.h
+
+#pragma warning disable 0649
+        struct ElfHeader
+        {
+            byte EI_MAG0;
+            byte EI_MAG1;
+            byte EI_MAG2;
+            byte EI_MAG3;
+
+            public bool IsValid()
+            {
+                return EI_MAG0 == 0x7f &&
+                       EI_MAG1 == 0x45 &&
+                       EI_MAG2 == 0x4C &&
+                       EI_MAG3 == 0x46;
+            }
+
+
+        }
+#pragma warning restore 0649
+
+        public static bool IsElfImage(string filePath)
+        {
+            using (BinaryReader reader = new BinaryReader(File.OpenRead(filePath)))
+            {
+                if (reader.BaseStream.Length < 16) // EI_NIDENT = 16
+                {
+                    return false;
+                }
+
+                byte[] eIdent = reader.ReadBytes(4);
```

Comment:
Newbie C# question: can you read directly on a `struct ElfHeader` and just return `header.IsValid()`?

Reply:
Binary Reader/Writer doesn't support structs (even explicit layout). 
I looked through the docs, but couldn't find a way to do it. 

So, I decided to just read the 4 bytes; we anyway didn't have the full header structure coded in this class -- just enough to check the magic bytes in the beginning.